app/scripts/views/settings/devices.js

```diff
@@ -0,0 +1,133 @@
+  var Url = require('lib/url');
+
+  var DEVICE_REMOVED_ANIMATION_MS = 150;
+  var DEVICES_SUPPORT_URL = 'https://support.mozilla.org/kb/fxa-managing-devices';
+  var FIREFOX_DOWNLOAD_LINK = 'https://www.mozilla.org/firefox/new/?utm_source=accounts.firefox.com&utm_medium=referral&utm_campaign=fxa-devices';
+  var FORCE_DEVICE_LIST_VIEW = 'forceDeviceList';
+
+  var View = FormView.extend({
+    template: Template,
+    className: 'devices',
+    viewName: 'settings.devices',
+
+    initialize: function (options) {
+      this._able = options.able;
+      this._devices = options.devices;
+
+      // An empty Devices instance is created to render the initial view.
+      // Data is only fetched once the panel has been opened.
+      if (! this._devices) {
+        this._devices = new Devices([], {
+          notifier: options.notifier
+        });
+      }
+
+      var devices = this._devices;
+      devices.on('add', this._onDeviceAdded.bind(this));
+      devices.on('change:name', this._onDeviceRenamed.bind(this));
+      devices.on('remove', this._onDeviceRemoved.bind(this));
+    },
+
+    context: function () {
+      console.log(this._isPanelEnabled());
+      return {
+        devices: this._devices.toJSON(),
+        devicesSupportUrl: DEVICES_SUPPORT_URL,
+        isPanelEnabled: this._isPanelEnabled(),
+        isPanelOpen: this.isPanelOpen(),
+        linkAndroid: FIREFOX_DOWNLOAD_LINK,
+        linkIOS: FIREFOX_DOWNLOAD_LINK,
+        linkLinux: FIREFOX_DOWNLOAD_LINK,
+        linkOSX: FIREFOX_DOWNLOAD_LINK,
+        linkWindows: FIREFOX_DOWNLOAD_LINK
+      };
+    },
+
+    events: {
+      'click .device-disconnect': preventDefaultThen('_onDisconnectDevice'),
+      'click .devices-refresh': preventDefaultThen('_onRefreshDeviceList')
+    },
+
+    _isPanelEnabled: function () {
+      return this._able.choose('deviceListVisible', {
+        forceDeviceList: Url.searchParam(FORCE_DEVICE_LIST_VIEW, this.window.location.search)
+      });
+    },
+
+    _onDeviceAdded: function () {
+      this.render();
+    },
+
+    _onDeviceRenamed: function (device) {
+      this.render();
+      var successMessage = Strings.interpolate(t('`%(previousName)s` renamed to `%(newName)s`'), {
+        newName: device.get('name'),
+        previousName: device.previous('name')
+      });
+
+      this.parentView.displaySuccess(successMessage);
+    },
+
+    _onDeviceRemoved: function (device) {
+      var id = device.get('id');
+      var self = this;
+      $('#' + id).slideUp(DEVICE_REMOVED_ANIMATION_MS, function () {
+        // re-render in case the last device is removed and the
+        // "no registered devices" message needs to be shown.
+        self.render();
+      });
+    },
+
+    _onDisconnectDevice: function (event) {
+      var deviceId = $(event.currentTarget).attr('data-id');
+      this._destroyDevice(deviceId);
+    },
+
+    _onRefreshDeviceList: function () {
+      this._fetchDevices();
```

Comment:
TODO: _onRefreshDeviceList should be called if there is a "device_refresh" event and ALSO IF PANEL IS OPEN. if panel is closed, ignore this event.